### PR TITLE
all: add report of failures at end

### PR DESCRIPTION
**Pull request by chrisyxlee**
Screen Shot 2023-05-11 at 19 53 23
Screen Shot 2023-05-11 at 19 53 39

https://github.com/wilsonehusin/smush/issues/4

## Changes

* `cmd/looper/main.go`: Fix a small inconsistency between format string and type
* `command.go`: Pass more information through channel to log a report at the end.
* `demo.yaml` -> `demo_fail.yaml`: Rename file to reflect desired output.
* `demo_pass.yaml`: New test case to see what will be printed on success.
* `log.go`: Don't pass error when there's 0% chance for an error. Change colors to reserve for run report.

### CODE REVIEW

File: command.go [hunk 1]
```diff
@@ -71,58 +79,112 @@ func (c *Command) Run(ctx context.Context, stdout, stderr io.Writer) error {
...
 	leftpad := 0
 	for _, command := range commands {
 		label := command.Label()
 		if len(label) > leftpad {
 			leftpad = len(label)
 		}
 	}
+	if leftpad < 3 {
+		leftpad = 3
+	}
```
Comment (wilsonehusin) [inline command.go]: we can probably just initialize the variable with 3 before the loop yeah?

```go
// Set minimum to 3, matching the anchor used for non-program logs.
leftpad := 3
```

**Review by wilsonehusin — APPROVED** [review]
❤️

File: command.go [hunk 2]
```diff
@@ -71,58 +79,112 @@ func (c *Command) Run(ctx context.Context, stdout, stderr io.Writer) error {
...
+	// Buffer the report so that all threads can exit first.
+	var report bytes.Buffer
+	report.WriteRune('\n')
+	report.WriteRune('\n')
+
+	rw := &Logger{
+		w:      &report,
+		prefix: []byte(fmt.Sprintf("%*s |> ", leftpad, "***")),
+	}
+	color.New(color.Bold, color.BgBlack, color.FgHiBlack).Fprint(rw, "RUN REPORT")
```
Comment (wilsonehusin) [inline command.go]: i have mixed feelings about the background color and text color combo here, feels like there isn't enough contrast?

Comment (chrisyxlee) [reply command.go]: Yeah, I'm not in love with this color combo either -- 

Basically I was trying to figure out how to handle when people use white terminal background vs. black terminal background -- maybe I'll do some research to see if you can determine that just by reading in some environment variable or something

File: command.go (hunk 1 above)
Comment (chrisyxlee) [reply command.go]: LOL
I saw the section and never actually read the code inside the for loop 😆 
Completely missed this part: 

```go
		if len(label) > leftpad {
			leftpad = len(label)
		}
```

File: command.go (hunk 2 above)
Comment (wilsonehusin) [reply command.go]: yeah the terminal background part is difficult. I think using bold font without the color alteration would be good enough for this? What do you think?

Comment (chrisyxlee) [reply command.go]: Simplified to just using bold!

**Pull request merged**